YesGraph/yesgraph-superwidget
# Misc updates

#### What’s this PR do?
- Fixes a bug in the deploy process, where minified versions weren't deploying.
- Adds testing to the deploy process
- Adds [gulp-autoprefixer](https://www.npmjs.com/package/gulp-autoprefixer) to the build process
 - Automatically adds prefixes (`-webkit-`, `-moz-`, etc.) to our css files, to make the styling more consistent across browsers.
- Adds sourcemaps for the minified javascript files.
 - It's difficult to debug Sentry errors when the traceback shows only minified code. With the new sourcemaps, Sentry should be able to automatically show tracebacks with _unminified_ code.
- Logs "breadcrumbs" to Sentry
 - These "breadcrumbs" will show up in Sentry error tracebacks, which should help us understand the sequence of events leading up to each error.
- Fixes a bug in the SDK, so that it doesn't try to send an empty array of analytics events.
- Fixes a bug in the SDK, by removing `Object.assign`
 - This method doesn't exist in many browsers, because it's new in ECMAScript 6

#### How should this be manually tested?
- To test the updates to the build process, run ` gulp build; ls dist/*.map`. This should show the names of the two new sourcemap files:
```
dist/yesgraph-invites.min.js.map
dist/yesgraph.min.js.map
```
- To test the updates to the code itself, run `gulp clean; npm test`


#### What are the relevant tickets?
Sourcemaps: https://app.asana.com/0/59202558034519/145112974620504
Testing: https://app.asana.com/0/59202558034519/145191540747963
